api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -942,6 +942,7 @@ message Cluster {
   // "envoy.filters.network.thrift_proxy". See the extension's documentation for details on
   // specific options.
   // [#next-major-version: make this a list of typed extensions.]
+  // [#extension-category: envoy.upstream_options]
```

Comment:
nit: consider renaming to upstream_protocol_options (as this is only for protocol options)

Reply:
That's essentially I did in the first pass, and wbpcode said no because in the code the category is this string. We could potentially change *both* things, but I don't even know what the purpose of category() [in code](https://github.com/envoyproxy/envoy/blob/41d8ccc79c4b98104475b55b5639567b14466190/source/extensions/upstreams/tcp/config.h#L46) is so I'm wary of touching it.

Reply:
I prefer to keep the doc be consisitant with code. But it's fine if we can change both. I also think `upstream_protocol_options` is better than `upstream_options`.

AFAIK, category() doesn't effect any logic in our code base? But I am not 100% whether the third-party forks will use it in some unexpected way.
I think you can throw this at maintainer group to see if anyone will object changing the category directly.

Reply:
Yeah, I just think that the `category()` returned value could be modified as well (but that's just a nit, and can be done in a followup PR if decided).

I agree that it will probably won't matter.